apps/abstraxion-dashboard/components/AbstraxionWallets/index.tsx

```diff
@@ -122,7 +84,67 @@ export const AbstraxionWallets = () => {
     } finally {
       setIsGeneratingNewWallet(false);
     }
-  };
+  }, [
+    apiUrl,
+    session_jwt,
+    session_token,
+    setIsGeneratingNewWallet,
+    startPolling,
+    setFetchingNewWallets,
+    setAbstraxionError,
+  ]);
+
+  useEffect(() => {
+    async function onStartup() {
+      if (!loading && data && !previousData && !isGeneratingNewWallet) {
+        if (
+          data?.smartAccounts?.nodes.length === 0 &&
+          connectionType === "stytch"
+        ) {
+          await handleJwtAALoginOrCreate();
+        }
+      }
+    }
+
+    onStartup();
+  }, [
+    loading,
+    data,
+    connectionType,
+    handleJwtAALoginOrCreate,
+    isGeneratingNewWallet,
+    previousData,
+  ]);
+
+  useEffect(() => {
+    if (previousData && data !== previousData) {
+      stopPolling();
+      setFetchingNewWallets(false);
+    }
+  }, [data, previousData, stopPolling]);
+
+  useEffect(() => {
```

Comment:
Is there any performance impact having several `useEffect` hooks in one component?

Reply:
There is. I didn't add any, but rather move them down and add the appropriate dependencies. Ideally we don't have too many useEffect hooks but given the more complex, autonomous, nature of this particular view, it's a bit hard to escape